core/src/main/java/io/undertow/server/handlers/URLDecodingHandler.java

```diff
@@ -57,36 +59,57 @@ public URLDecodingHandler(final HttpHandler next, final String charset) {
     public void handleRequest(final HttpServerExchange exchange) throws Exception {
         boolean decodeDone = exchange.getConnection().getUndertowOptions().get(UndertowOptions.DECODE_URL, true);
         if (!decodeDone) {
```

Comment:
Thoughts on adding an attachment key here so that we can avoid decoding values that have already been decoded? Otherwise we risk turning `%253E` into `>` rather than `%3E`.

For example, I'd like this to do the right thing: `new URLDecodingHandler(new URLDecodingHandler(next, "UTF-8"), "UTF-8")`

I'll file a ticket for discussion

Reply:
Ticket: https://issues.jboss.org/browse/UNDERTOW-1479
Proposed fix: https://github.com/undertow-io/undertow/commit/2f8911316f077aa7d42e52d049c8516f41c0c03e

Reply:
Added the commit to this branch. The check includes both the check for `UndertowOptions.DECODE_URL` and the new marker.